cpp/csp/python/Common.h

```diff
@@ -7,6 +7,22 @@
 
 #define INIT_PYDATETIME if( !PyDateTimeAPI ) { PyDateTime_IMPORT; }
 
+// NumPy 2.0 Migration
+#include <numpy/numpyconfig.h>
+
+#if NPY_ABI_VERSION >= 0x02000000
+// Define helper for anything that can't
+// be handled by the below helper macros
+#define CSP_NUMPY_2
```

Comment:
not really following why this is needed at all since it looks like we're only be compiling with numpy >= 2 after this change

Reply:
see https://github.com/Point72/csp/pull/328/files#r1686672275